fuse/ipns/ipns_unix.go

```diff
@@ -140,7 +140,7 @@ func (s *Root) Lookup(ctx context.Context, name string) (fs.Node, error) {
 	}
 
 	// other links go through ipns resolution and are symlinked into the ipfs mountpoint
-	resolved, err := s.Ipfs.Namesys.Resolve(s.Ipfs.Context(), name)
+	resolved, err := s.Ipfs.Namesys.Resolve(s.Ipfs.Context(), name, 0)
```

Comment:
On Thu, May 14, 2015 at 09:33:18PM -0700, Juan Batiz-Benet wrote:
> > -	resolved, err := s.Ipfs.Namesys.Resolve(s.Ipfs.Context(), name)
> > +	resolved, err := s.Ipfs.Namesys.Resolve(s.Ipfs.Context(), name, 0)
> 
> why is this resolving with depth 0? nothing should resolve
> infinitely. yeah this is what i want to avoid

Because the calling code doesn't care about how many steps it takes,
it just wants a resolved path.  If we want to limit the resolution
depth as a policy issue, we just need a way to inject the policy limit
into namesys/base.go's resolve.  That's independent of the calling
code.  In my not-pushed-yet reroll, I've gone with the hard-coded 32
limit mentioned here [1].  Commit looks like:

   // resolve is a helper for implementing Resolver.Resolve using resolveOnce.
   func resolve(ctx context.Context, r resolver, name string, depth int, prefixes ...string) (path.Path, error) {
  +       id depth == 0 {
  +               depth = 32
  +       }
          for {

And I'm punting a serious place to configure that default depth to a
follow-up PR (as mentioned in [1]).

[1]: https://github.com/ipfs/go-ipfs/pull/1208#issuecomment-101526369